ibis/backends/bigquery/__init__.py

```diff
@@ -147,6 +151,136 @@ def _register_in_memory_table(self, op: ops.InMemoryTable) -> None:
             )
             load_job.result()
 
+    def _read_file(
+        self,
+        path: str | Path,
+        *,
+        table_name: str | None = None,
+        job_config: bq.LoadJobConfig,
+    ) -> ir.Table:
+        self._make_session()
+
+        if table_name is None:
+            table_name = util.gen_name(f"bq_read_{job_config.source_format}")
+
+        dataset = self._session_dataset.dataset_id
+        project = self._session_dataset.project
+
+        self.drop_table(table_name, schema=dataset, database=project, force=True)
```

Comment:
FYI: This step isn't required if we set `job_config.write_disposition = "WRITE_TRUNCATE"`, but I see we need append-mode below for multiple files. Might be worth a comment explaining that.

Reply:
Thanks for the review! Indeed, I took the `WRITE_TRUNCATE` approach first before tinkering with loading multiple files concurrently.

I'll add a comment.

Reply:
Actually, I think since it's possible to cause data loss this way, I'm going to check whether the table exists first and error out.

Reply:
Reconsidering my last statement, since this is all going in the anonymous dataset overwriting things is less risky, since it's hidden from most public-facing APIs/applications.